app/models/coronavirus/announcement.rb

```diff
@@ -18,13 +36,20 @@ def set_parent_positions
     page.make_announcement_positions_sequential
   end
 
-  def published_at_format
-    unless published_at.is_a?(Time) && valid_year?
-      errors.add(:published_at, "must be a valid date")
-    end
+  def parsed_published_at_hash
+    day, month, year = published_at_hash.values_at("day", "month", "year").map(&:to_i)
+    Date.strptime("#{day}-#{month}-#{year}", "%d-%m-%Y")
+  rescue ArgumentError
+    nil
   end
 
-  def valid_year?
-    published_at.past? && published_at.year > 1950
+  def valid_published_at
+    if published_at_hash && !published_at
+      errors.add(:published_at, "must be a valid date")
+    elsif published_at && published_at.year < 2000
```

Comment:
Nitpick: could this also be `published_at&.year < 2000` to be consistent with the clause below?

Reply:
Sadly not. With the safe navigation operator you end up risking `nil < 2000`  which risks a `NoMethodError (undefined method `<' for nil:NilClass)` exception.